vendor/bat-native-ads/src/bat/ads/internal/ml/model/linear/linear_unittest.cc

```diff
@@ -0,0 +1,138 @@
+  // Arrange
+  std::map<std::string, data::VectorData> weights = {
+      {"class_1", data::VectorData(std::vector<double>{1.0, 0.0, 0.0})},
+      {"class_2", data::VectorData(std::vector<double>{0.0, 1.0, 0.0})},
+      {"class_3", data::VectorData(std::vector<double>{0.0, 0.0, 1.0})}};
+
+  std::map<std::string, double> biases = {
+      {"class_1", 0.5}, {"class_2", 0.25}, {"class_3", 1.0}};
+
+  model::Linear linear_biased(weights, biases);
+  data::VectorData avg_vector(std::vector<double>{1.0, 1.0, 1.0});
+
+  // Act
+  PredictionMap res = linear_biased.Predict(avg_vector);
+
+  // Assert
+  EXPECT_TRUE(res["class_3"] > res["class_1"] &&
+              res["class_3"] > res["class_2"] &&
+              res["class_1"] > res["class_2"]);
+}
+
+TEST_F(BatAdsLinearModelTest, BinaryClassifierPredictionTest) {
+  // Arrange
+  std::map<std::string, data::VectorData> weights = {
+      {"the_only_class", data::VectorData(std::vector<double>{0.3, 0.2, 0.25})},
+  };
+
+  std::map<std::string, double> biases = {
+      {"the_only_class", -0.45},
+  };
+
+  model::Linear linear(weights, biases);
+  data::VectorData data_vector_0(std::vector<double>{1.07, 1.52, 0.91});
+  data::VectorData data_vector_1(std::vector<double>{1.11, 1.63, 1.21});
+
+  // Act
+  PredictionMap res_0 = linear.Predict(data_vector_0);
+  PredictionMap res_1 = linear.Predict(data_vector_1);
+
+  // Assert
+  ASSERT_EQ(res_0.size(), static_cast<size_t>(1));
+  ASSERT_EQ(res_1.size(), static_cast<size_t>(1));
+
+  EXPECT_TRUE(res_0["the_only_class"] < 0.5 && res_1["the_only_class"] > 0.5);
+}
+
+TEST_F(BatAdsLinearModelTest, TopPredictionsTest) {
+  // Arrange
+  std::map<std::string, data::VectorData> weights = {
+      {"class_1", data::VectorData(std::vector<double>{1.0, 0.5, 0.8})},
+      {"class_2", data::VectorData(std::vector<double>{0.3, 1.0, 0.7})},
+      {"class_3", data::VectorData(std::vector<double>{0.6, 0.9, 1.0})},
+      {"class_4", data::VectorData(std::vector<double>{0.7, 1.0, 0.8})},
+      {"class_5", data::VectorData(std::vector<double>{1.0, 0.2, 1.0})}};
+
+  std::map<std::string, double> biases = {{"class_1", 0.21},
+                                          {"class_2", 0.22},
+                                          {"class_3", 0.23},
+                                          {"class_4", 0.22},
+                                          {"class_5", 0.21}};
+
+  model::Linear linear_biased(weights, biases);
+  data::VectorData point_1(std::vector<double>{1.0, 0.99, 0.98, 0.97, 0.96});
+  data::VectorData point_2(std::vector<double>{0.83, 0.79, 0.91, 0.87, 0.82});
+  data::VectorData point_3(std::vector<double>{0.92, 0.95, 0.85, 0.91, 0.73});
+
+  // Act
+  PredictionMap res_1 = linear_biased.TopPredictions(point_1);
+  PredictionMap res_2 = linear_biased.TopPredictions(point_2, 2);
+  PredictionMap res_3 = linear_biased.TopPredictions(point_3, 1);
+
+  // Assert
+  ASSERT_EQ(res_1.size(), static_cast<size_t>(5));
```

Comment:
nit: expected value on the left, actual value on the right

Reply:
Done